src/Illuminate/Console/Command.php

```diff
@@ -189,7 +198,94 @@ public function run(InputInterface $input, OutputInterface $output)
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->laravel->call([$this, 'handle']);
+        if ($this->hasValidInput()) {
+            return $this->laravel->call([$this, 'handle']);
+        }
+
+        return $this->displayValidationErrors();
+    }
+
+    /**
+     * Check if the argument and option values are valid.
+     *
+     * @return bool
+     */
+    protected function hasValidInput()
+    {
+        if (empty($rules = $this->rules())) {
+            return true;
+        }
+
+        try {
+            $this->validator = $this->validator ?? $this->makeValidator();
+        } catch (BindingResolutionException $e) {
+            return true;
+        }
+
+        $this->validator->make(
+            array_merge($this->arguments(), $this->options()),
+            $rules,
+            $this->messages(),
+            $this->attributes()
+        );
+
+        return ! $this->validator->fails();
+    }
+
+    /**
+     * Resolve the command input validator.
+     *
+     * @return \Illuminate\Contracts\Validation\Factory
+     */
+    protected function makeValidator()
+    {
+        return $this->laravel->make(ValidatorFactory::class);
+    }
+
+    /**
+     * Define the rules used for validation of arguments and options.
+     *
+     * @return array
+     */
+    protected function rules()
+    {
+        return [];
+    }
+
+    /**
+     * Get the custom error messages for console command validation.
+     *
+     * @return array
+     */
+    protected function messages()
+    {
+        return [];
+    }
+
+    /**
+     * Get the custom attribute names for console command validation.
+     *
+     * @return array
+     */
+    protected function attributes()
+    {
+        return [];
+    }
+
+    /**
+     * Display the validation error messages for invalid arguments and options.
+     *
+     * @return int
+     */
+    protected function displayValidationErrors()
+    {
+        $this->info('Command input validation failed. Please check the error messages below:');
```

Comment:
you probably will want to show errors `just` if there is any.

Reply:
As far as I see there will always be at least one error when this is called since it gets called only if the validation fails.